model/src/pyrenew/metaclass.py

```diff
@@ -95,14 +95,80 @@ def _assert_sample_and_rtype(
 class RandomVariable(metaclass=ABCMeta):
     """
     Abstract base class for latent and observed random variables.
+
+    Notes
+    -----
+    `RandomVariable`s in `pyrenew` can be time-aware, meaning that they can
+    have a `timeseries_start` and `timeseries_unit` attribute. These attributes
+    are expected to be used internally mostly for tasks including padding,
+    alignment of time series, and other time-aware operations.
+
+    Both attributes give information about the output of the `sample()` method,
+    in other words, the relative time units of the returning value.
+
+    Attributes
+    ----------
+    timeseries_start : int
+        The start of the time series.
+    timeseries_unit : int
+        The unit of the time series, e.g., 1: Daily, 7: Weekly.
     """
 
+    timeseries_start: int = None
+    timeseries_unit: int = None
+
+    def __init_subclass__(cls) -> None:
+        """Initializes the subclass with default values."""
+        cls.timeseries_unit = 1
```

Comment:
I am setting a default value instantiated during construction for `timeseries_unit`. I am not doing the same for `timeseries_start` because if we, for instance, set it to be zero, we cannot check whether that was set on purpose, or the user forgot to explicitly set the start value, what are your thoughts? After writing this, I am inclining to say no defaults, set all to `None` and then we assume that the `timeseries_unit` equals 1 by default across the package.

Reply:
1. Why do this via an `__init_subclass__` method rather than via the default constructor?
2. Even at initialization, isn't it more futureproof to use the setting method (with validation) that you define below?

Reply:
1. B/c the `__init__` method may be instance specific, the `__init_subclass__` gives the flexibility of always doing something during instantiation, regardless of the `__init__` method of the instance.
2. Could be! I was wondering that. Furthermore, I wonder if we want `timeseries_unit` to be 1 by default. Maybe, we could just make it None. What do you think?

Reply:
I prefer default `None` for both attributes at the metaclass level, which also obviates using `__initsubclass__`

Reply:
Suggestion below.